src/sidebar/services/groups.js

```diff
@@ -89,23 +95,28 @@ function groups(
     // link to an annotation in that group.
     const nonWorldGroups = groups.filter(g => g.id !== '__world__');
 
-    if (!directLinkedAnnotationId) {
+    if (!directLinkedAnnotationId && !directLinkedGroupId) {
       return Promise.resolve(nonWorldGroups);
     }
 
-    return api.annotation
-      .get({ id: directLinkedAnnotationId })
+    // If the directLinkedGroupId or the directLinkedAnnotationId's group
+    // is the "Public" group show the "Public" group otherwise do not.
```

Comment:
If `directLinkedGroupId` is set to `__world__` then we always return `groups` rather than `nonWorldGroups` here. Therefore I think the code can be simplified by adding an early return in that case and leaving the rest of the function as it was before:

```
if (directLinkedGroupId === '__world__') {
  return Promise.resolve(groups);
}
```